### PR TITLE
sopel: 6.1.1 -> 6.2.1

### PR DESCRIPTION
**Pull request by mogorman**
version bump

### CODE REVIEW

**Comment by mogorman** [comment]
not sure what caused error? i only changed version. it works on my box?

**Comment by mogorman** [comment]
@FRidh do you know why python is breaking here?  I don't see how my changes could be doing this.  someone on irc told me i should update my fork to latest master, I tried that just now and it still fails on same syntax error

**Comment by mogorman** [comment]
it looks like its broken for everyone actually.  reading commit log of the python-packages doesnt seem like any mistake was made since my last commit

**Comment by FRidh** [comment]
Your patch builds fine. While you are at it, could you fix the testing for this package?
It apparently needs `pytest` to test the folder `test`.

**Comment by mogorman** [comment]
added pytest to buildinputs

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -18315,13 +18315,13 @@ in modules // {
...
-    name = "sopel-6.1.1";
+    name = "sopel-6.2.1";
 
     src = pkgs.fetchurl {
       url = "https://pypi.python.org/packages/source/s/sopel/${name}.tar.gz";
-      sha256 = "0nr2a88bkxg2593dd947qkh96g8j32q7pl7x9zvx4158h4bgx99y";
+      sha256 = "06m5clmg9x0bsnhvl5d75mskwqnxvkdd00p0dqnpwip9vmq6n8cz";
     };
-
+    BuildInputs = with self; [ pytest ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: typo

**Comment by FRidh** [comment]
unfortunately just adding the test runner as input is not sufficient. We also need to actually run it. In `python-packages.nix` you can see some examples how pytest is invoked.

**Comment by mogorman** [comment]
it runs test dir now

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -18315,17 +18314,21 @@ in modules // {
...
-    name = "sopel-6.1.1";
+    name = "sopel-6.2.1";
 
     src = pkgs.fetchurl {
       url = "https://pypi.python.org/packages/source/s/sopel/${name}.tar.gz";
-      sha256 = "0nr2a88bkxg2593dd947qkh96g8j32q7pl7x9zvx4158h4bgx99y";
+      sha256 = "06m5clmg9x0bsnhvl5d75mskwqnxvkdd00p0dqnpwip9vmq6n8cz";
     };
 
+    BuildInputs = with self; [ pytest ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: typo

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -956,8 +956,7 @@ in modules // {
     };
 
     checkPhase = ''
-      export PYTHONPATH=`pwd`/build/lib:$PYTHONPATH
-      ${python.interpreter} test/test_argparse.py
+      ${python.interpreter} test/test_db.py
     '';
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: I suppose this was unintentional?

**Comment by mogorman** [comment]
yes it was embarrising

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -18315,17 +18315,21 @@ in modules // {
...
     };
 
+    buildInputs = with self; [ pytest ];
     propagatedBuildInputs = with self; [ praw xmltodict pytz pyenchant pygeoip ];
 
     disabled = isPyPy || isPy26 || isPy27;
 
+    checkPhase = ''
+    ${python.interpreter} test/*.py
+    '';
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Generally, when we have pytest, we invoke the test runner using
```
checkPhase = ''
  pytest test
```
Sometimes the default `${python.interpreter} test` is sufficient. I just checked upstream to find out how they run their tests, and I noticed this [file](https://github.com/sopel-irc/sopel/blob/6.2.1/pytest_run.py). Maybe we should run that instead in the checkPhase?

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: anyway, as long as it runs the tests and succeeds

**Pull request merged**